CalPatRec/src/PhiClusterFinder_module.cc

```diff
@@ -0,0 +1,566 @@
+        if (_debug>2) printf("Phi min = %10.3f max : %10.3f\n",cluphimin,cluphimax);
+        return;
+      }
+    }
+    // Check to the right of the highest bin
+    int bincheckr=max_bin;
+    while(_hist1->GetBinContent(bincheckr)>=_threshold){
+      // if(_debug>3) std::cout<<"bincheckr = "<<_hist1->GetBinContent(bincheckr)<<"bincontent = "<<bincheckr<<"  "<<_hist1->GetBinCenter(bincheckr)<<std::endl;
+      if(bincheckr<nbx) bincheckr++;
+      else bincheckr = 1;
+    }
+    cluphimax = _hist1->GetXaxis()->GetBinCenter(bincheckr)+bin/2;
+    cluphimin = _hist1->GetXaxis()->GetBinCenter(bincheck )-bin/2;
+    if (_debug>3) printf("Phi min = %10.3f max : %10.3f\n",cluphimin,cluphimax);
+  }
+
+//------------------------------------------------------------------------------
+// Function to fill the time spectrum
+//-----------------------------------------------------------------------------
+  void PhiClusterFinder::fillTimeSpectrum(int ClustNo, const std::vector<StrawHitIndex>& ordchcol) {
+    _timespec.Reset();
+    for(unsigned istr=0; istr<ordchcol.size();++istr) {
+      int i = istr;
+      if(_clustno[i]==ClustNo){
+        ComboHit const& ch = (*_chcol)[istr];
+        float time = _t0calc.comboHitTime((*_chcol)[istr],_pitch);
+        _timespec.Fill(time,ch.nStrawHits());
+      }
+    }
+  }
+
+//-----------------------------------------------------------------------------------
+// Function to fill the time spectrum when the hit is not associated to a phi cluster
+//-----------------------------------------------------------------------------------
+  void PhiClusterFinder::fillTimeSpectrumAlt(const std::vector<StrawHitIndex>& ordchcol) {
+    _timespec.Reset();
+    for(unsigned istr=0; istr<ordchcol.size();++istr) {
+      ComboHit const& ch = (*_chcol)[istr];
+      float time = _t0calc.comboHitTime((*_chcol)[istr],_pitch);
+      _timespec.Fill(time,ch.nStrawHits());
+    }
+  }
+
+
+//------------------------------------------------------------------------------
+// Function to find the peak bin the time spectrum
+//-----------------------------------------------------------------------------
+  void PhiClusterFinder::findPeaks(TimeCluster& tc, const std::vector<StrawHitIndex>& ordchcol) {
+    int nbins = _timespec.GetNbinsX()+1;
+    // loop over spectrum to find peaks
+    std::vector<BinContent> bcv;
+    for (int ibin=1;ibin < nbins; ++ibin)
+      if (_timespec.GetBinContent(ibin) >= _ymin) bcv.push_back(make_pair(_timespec.GetBinContent(ibin),ibin));
+    std::sort(bcv.begin(),bcv.end(),[](const BinContent& x, const BinContent& y){return x.first > y.first;});
+    for (const auto& bc : bcv) {
+      float nsh(0.0);
+      float t0(0.0);
+      for (int ibin = std::max(1,bc.second-_npeak);ibin < std::min(nbins,bc.second+_npeak+1); ++ibin) {
+        nsh += _timespec.GetBinContent(ibin);
+        t0 += _timespec.GetBinCenter(ibin)*_timespec.GetBinContent(ibin);
+      }
+      t0 /= nsh;
+      if (nsh >= _minnsh){
```

Comment:
the way this nested loop is set up, it uses always the last entry of the vector `bcv` if `nsh >_minnsh`, so it doesn't take the highest one because `bcv` is previously sorted such that the highest bin is the first entry. Am I missing anything?

Reply:
Hi @gianipez I've cleaned up the module quite a bit. Removed the functions like assignHits, findPeak, fillTimeSpectrum, all of which are from the time when we were using the ComboHitCollection as input.